realm/realm-library/src/objectServer/java/io/realm/PermissionManager.java

```diff
@@ -705,12 +742,17 @@ public void onSuccess() {
                     managedOffer = managementRealm.where(PermissionOffer.class).equalTo("id", offerId).findFirstAsync();
                     RealmObject.addChangeListener(managedOffer, new RealmChangeListener<PermissionOffer>() {
                         @Override
-                        public void onChange(PermissionOffer permissionOffer) {
+                        public void onChange(final PermissionOffer permissionOffer) {
                             if (checkAndReportInvalidState()) {
                                 RealmObject.removeChangeListener(managedOffer, this);
                                 return;
                             }
-                            handleServerStatusChanges(permissionOffer, permissionOffer.getToken());
+                            handleServerStatusChanges(permissionOffer, new Runnable() {
```

Comment:
Again, each one of these anonymous classes is an entirely new class.  It might be worth creating a single named class for this, and just creating instances as you need them.  Lambdas will do a better job of this, when we can use them.